stream_alert_cli/terraform/generate.py

```diff
@@ -271,27 +272,39 @@ def generate_cluster(**kwargs):
     return cluster_dict
 
 
+def cleanup_old_tf_files():
+    """Cleanup old .tf files, these are now .tf.json files per Hashicorp best practices"""
+    for terraform_file in os.listdir('terraform'):
+        if terraform_file == 'variables.tf':
+            continue
+
+        if fnmatch(terraform_file, '*.tf'):
+            os.remove(os.path.join('terraform', terraform_file))
+
+
 def terraform_generate(**kwargs):
     """Generate all Terraform plans for the configured clusters.
 
     Keyword Args:
         config (dict): The loaded config from the 'conf/' directory
-        init (bool): Indicates if main.tf is generated for `terraform init`
+        init (bool): Indicates if main.tf.json is generated for `terraform init`
 
     Returns:
         bool: Result of cluster generating
     """
     config = kwargs.get('config')
     init = kwargs.get('init', False)
 
-    # Setup the main.tf file
-    LOGGER_CLI.debug('Generating cluster file: main.tf')
+    cleanup_old_tf_files()
+
+    # Setup the main.tf.json file
+    LOGGER_CLI.debug('Generating cluster file: main.tf.json')
     main_json = json.dumps(
         generate_main(init=init, config=config),
         indent=2,
         sort_keys=True
     )
-    with open('terraform/main.tf', 'w') as tf_file:
+    with open('terraform/main.tf.json', 'w') as tf_file:
```

Comment:
Thanks for this change and adding the logic to remove the old files - one question though:

Does the `terraform_clean` function need to be updated to remove cluster files based on their new extension as well?

Reply:
Yep, good call.  I'll move more of this logic into the `stream_alert_cli/terraform` package